### ISSUE TITLE
undefined index in update_control.class.php

### ISSUE THREAD

**Issue by andtrev** [CodeCabin/wp-live-chat-support#948]
A few users have come across an issue with undefined index notices when $checked_data->checked[$this->wplc_api_slug . '/' . $this->wplc_api_slug . '.php'] is accessed.

**Comment by NickDuncanZA** [comment]
We need more information on this. Next time we get a user experiencing this issue lets try dig a little and see what we find.

**Comment by andtrev** [comment]
I did attempt to patch this, it's just that the plugin isnt in $checked_data, in my patch I attempt to locate the plugin and get the version when it isn't in $checked_data, but I'm realizing now that one reason it might not be in that array is if it is installed in a different folder name than expected.

We should probably try and find the proper slug for the plugin being updated in the list of activated plug-ins rather than relying on the passed in slug value, which may be incorrect, before trying to get version information from the plugin file itself as a last ditch effort. Active plug-ins are in active_plugins in the options table so we should be able to search that easily to find the full correct slug.

**Comment by DylanCodeCabin** [comment]
Yes I agree, we should look into the way we find the slug, we have seen issues like this with updates. More specifically, when the slug does not match exactly, the update is refused. API key is then listed as invalid. 

Makes more sense for us to check if the slug matches partially, instead of exactly.
